controllers/machinehealthcheck_controller.go

```diff
@@ -252,12 +254,57 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	for _, t := range unhealthy {
 		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
 
-		if annotations.IsPaused(cluster, t.Machine) {
-			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+		if m.Spec.RemediationTemplate != nil {
+			// If external remediation request already exists,
+			// return early
+			if r.externalRemediationRequestExists(ctx, m, t) {
+				logger.Info("External Remediation Request for", t.Machine.Name, " exists")
+				return ctrl.Result{}, nil
+			}
+
+			cloneOwnerRef := &metav1.OwnerReference{
+				APIVersion: clusterv1.GroupVersion.String(),
+				Kind:       t.Machine.Kind,
+				Name:       t.Machine.Name,
+				UID:        t.Machine.UID,
+			}
+
+			from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Remediation template not found %q", m.Spec.RemediationTemplate)
+			}
+
+			generateTemplateInput := &external.GenerateTemplateInput{
+				Template:    from,
+				TemplateRef: m.Spec.RemediationTemplate,
+				Namespace:   t.Machine.Namespace,
+				ClusterName: t.Machine.ClusterName,
+				OwnerRef:    cloneOwnerRef,
+				Labels:      nil,
+			}
+			to, err := external.GenerateTemplate(generateTemplateInput)
+			if err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to create template for remediation request %q", m.Spec.RemediationTemplate)
+			}
+
+			to.SetName(t.Machine.Name)
+
+			// Create the external clone.
+			if err := r.Client.Create(context.Background(), to); err != nil {
+				return ctrl.Result{}, errors.Wrapf(err, "Failed to create remediation request for machine %q", t.Machine.Name)
+			}
+
+			logger.Info("Target has failed health check, marking for external remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+
 		} else {
-			logger.Info("Target has failed health check, marking for remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
-			conditions.MarkFalse(t.Machine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason, clusterv1.ConditionSeverityWarning, "MachineHealthCheck failed")
+			if annotations.IsPaused(cluster, t.Machine) {
+				logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+			} else {
+				logger.Info("Target has failed health check, marking for remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+				conditions.MarkFalse(t.Machine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason, clusterv1.ConditionSeverityWarning, "MachineHealthCheck failed")
+			}
```

Comment:
Should this be in this PR? Might be worth to pull the support for paused machined in a different PR with related tests

Reply:
> Should this be in this PR? Might be worth to pull the support for paused machined in a different PR with related tests

That's fine for me. We can remove testing of paused machine here and create new PR for supporting it.

Reply:
@ncdc @wfernandes here is earlier discussion earlier with @vincepri about Pause annotation and its tests.

Reply:
Unless I'm misreading my local copy of the code, we currently have support for skipping remediating paused machines. Let's not remove it?

Reply:
It was added in #3444

Reply:
@ncdc Done. I reversed the changes and returned the tests

Reply:
So just to understand the logic better since we now have two `if annotations.IsPaused` checks (line 261 and 313).

**If a machine is unhealthy and is also marked as Paused, do we want to do the external remediation?** If we don't need to do external remediation for a machine that's paused then maybe we can clean up the logic a little bit. Maybe do something like,
```golang
if annotations.IsPaused(cluster, t.Machine) {
  // Print out log message - Machine has failed health check, but machine is paused so skipping remediation
} else {
  if m.Spec.RemediationTemplate != nil {
    // Do external remediation stuff
    // DO WE NEED ANY CONDITIONS HERE? I thought we discussed something about ExternalRemediationTemplateAvailable
  } else {
    // Print out log message - Target has failed health check, marking for remediation
    // Mark MachineOwnerRemediatedCondition as false
  }
}
```

Maybe we can add a test for this behavior? WDYT?

Reply:
Sorry, this was totally my bad and a mistake. I would delete from line 261 and keep it in 313.

Reply:
Ok. So we are going to do external remediation for an unhealthy machine even if it is marked as Paused. Thanks for clarifying.